File: frontend/src/types/prediction.ts
export interface ImageInfo {
  filename: string;
  content_type: string;
  size_bytes: number;
  width: number;
  height: number;
}

export interface ClassPrediction {
  class_name: string;
  probability: number;
  threshold: number;
  is_predicted: boolean;
}

export interface PredictionResponse {
  image: ImageInfo;
  predicted_labels: ClassPrediction[];
  all_scores: ClassPrediction[];
  top_k: ClassPrediction[];
  model: Record<string, unknown>;
  message: string | null;
}

export interface ModelInfoResponse {
  available: boolean;
  architecture: string | null;
  num_classes: number | null;
  classes: string[];
  image_size: number | null;
  thresholds: Record<string, number>;
  validation_metrics: Record<string, unknown>;
  message: string | null;
}

File: backend/app/schemas/prediction.py
from __future__ import annotations

from typing import Any

from pydantic import BaseModel


class ImageInfo(BaseModel):
    filename: str
    content_type: str
    size_bytes: int
    width: int
    height: int


class ClassPrediction(BaseModel):
    class_name: str
    probability: float
    threshold: float
    is_predicted: bool


class PredictionResponse(BaseModel):
    image: ImageInfo
    predicted_labels: list[ClassPrediction]
    all_scores: list[ClassPrediction]
    top_k: list[ClassPrediction]
    model: dict[str, Any]
    message: str | None = None
